### PR TITLE
User API Updates

### PR DESCRIPTION
**Pull request by HankG**
This is feature complete and ready **except** for the contacts return method. That method is probably extraneous since the only user whose contacts one can get is your own and that is already supported in the Contacts API endpoint.  If you agree I'm going to rip it out. This PR right now is just so for review of the rest.

### CODE REVIEW

**Comment by HankG** [comment]
Probably want a contacts endpoint but one that only works on the default user now the Contacts API is really for probing the aspect's contacts.

File: config/routes.rb [hunk 1]
```diff
@@ -234,7 +234,14 @@
       resources :messages, only: %i[index create]
     end
 
+    patch "user" => "users#update"
+    get "user" => "users#show"
     resources :tag_followings, only: %i[index create destroy]
+    resources :users, only: %i[show update] do
```
Comment (frankrousseau) [inline config/routes.rb]: are you sure that update is needed here?

**Comment by HankG** [comment]
This is ready for full review now.

**Comment by frankrousseau** [comment]
Ok. I'm a bit busy. But I will do my best to review it soon.

File: app/services/aspects_membership_service.rb [hunk 2]
```diff
@@ -43,6 +43,12 @@ def contacts_in_aspect(aspect_id)
     ).includes(person: :profile).order(order)
   end
 
+  def all_contacts
+    order = ["contact_id IS NOT NULL DESC", "profiles.first_name ASC", "profiles.last_name ASC",
+             "profiles.diaspora_handle ASC"]
+    @user.contacts.includes(person: :profile)
```
Comment (frankrousseau) [inline app/services/aspects_membership_service.rb]: It's like you don't use order here.

File: spec/integration/api/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "Single User" do
+      it "succeeds with public user not in Aspect" do
+        alice.profile[:public_details] = true
+        alice.profile.save
+        get(
+          "/api/v1/users/#{alice.guid}",
```
Comment (frankrousseau) [inline spec/integration/api/users_controller_spec.rb:40]: Why don't you use Rails path helpers here?

File: spec/integration/api/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+    expect(post_person["diaspora_id"]).to eq(user.diaspora_handle)
+    expect(post_person["name"]).to eq(user.name)
+    expect(post_person["avatar"]).to eq(user.profile.image_url)
+  end
+
+  def confirm_photos(photos)
+    photos.each do |photo|
+      expect(photo.has_key?("guid")).to be_truthy
+      expect(photo["dimensions"].has_key?("height")).to be_truthy
+      expect(photo["dimensions"].has_key?("height")).to be_truthy
```
Comment (frankrousseau) [inline spec/integration/api/users_controller_spec.rb]: This test is the same as the one above.

File: config/routes.rb (hunk 1 above)
Comment (HankG) [reply config/routes.rb]: I believe so.  Let me see if it works without it.

File: spec/integration/api/users_controller_spec.rb (hunk 3 above)
Comment (HankG) [reply spec/integration/api/users_controller_spec.rb:40]: I couldn't get them to generate that path no matter how hard I tried.  It happened on another endpoint as well.

File: app/services/aspects_membership_service.rb (hunk 2 above)
Comment (HankG) [reply app/services/aspects_membership_service.rb]: whoops...great catch...

File: config/routes.rb (hunk 1 above)
Comment (HankG) [reply config/routes.rb]: I was able to take it out.

**Comment by HankG** [comment]
This is squashed and I believe ready for merge if you like the way I addressed the comments...

File: spec/integration/api/users_controller_spec.rb (hunk 3 above)
Comment (frankrousseau) [reply spec/integration/api/users_controller_spec.rb:40]: Ok don't worry. I just wanted to be sure.

**Comment by frankrousseau** [comment]
Need a rebase. It looks like I didn't merge the PRs in the right order. If there is any, tell me :)

**Comment by HankG** [comment]
> Need a rebase. It looks like I didn't merge the PRs in the right order. If there is any, tell me :)

Because these are editing routes.rb and the en.yml files every time you merge in a PR this is going to happen.  We can just walk down them one at a time...

**Comment by frankrousseau** [comment]
Ok let's go for merging them one by one.

**Comment by HankG** [comment]
OK sounds good...this one is ready. I'll do the "search" next since it is a one liner and then I'll merge all that into "photos" which needs a full review before squash and merge.

**Pull request merged**